libresapi/src/api/FileSharingHandler.h

```diff
@@ -48,10 +55,13 @@ class FileSharingHandler: public ResourceRouter
 
 	void handleDownload(Request& req, Response& resp);
 
-	StateToken mStateToken;
+	StateToken mLocalDirStateToken;
+	StateToken mRemoteDirStateToken;
 	StateTokenServer* mStateTokenServer;
+	RsMutex mMtx;
 
 	RsFiles* mRsFiles;
+	RsNotify* mNotify;
```

Comment:
It is preferable to use a reference instead of a pointer if possible (aka. if it is always valid and doesn't change between initialization and destruction)